src/Storages/Kafka/KafkaBlockInputStream.cpp

```diff
@@ -189,6 +212,15 @@ Block KafkaBlockInputStream::readImpl()
                 }
                 virtual_columns[6]->insert(headers_names);
                 virtual_columns[7]->insert(headers_values);
+                if (put_error_to_stream)
+                {
+                    auto payload = buffer->currentPayload();
+                    virtual_columns[8]->insert(payload);
```

Comment:
Not sure if we need to push the row message if there was no error (it will multiply the size of data for all valid rows).